pkg/apis/clusterregistry/v1alpha1/types.go

```diff
@@ -30,6 +30,62 @@ type Cluster struct {
 	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
 	// +optional
 	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// State is thestate of the cluster. This is not a specification, and is not
+	// meant to be used by actively-reconciling controllers; it is also not
+	// a status, as it contains fields that do not necessarily describe the
+	// status of the cluster, and is not necessarily updated by an active
+	// controller.
+	State ClusterState
+}
+
+// ClusterState contains the state of a cluster.
+type ClusterState struct {
+	KubeAPIServer KubeAPIServer `json:"kube_api_server,omitempty" protobuf:"bytes,1,opt,name=kube_api_server"`
+	AuthInfo      AuthInfo      `json:"auth_info,omitempty" protobuf:"bytes,2,opt,name=auth_info"`
```

Comment:
If these fields are optional, they should be pointers if they do not have a meaningful zero type.  I think `KubeAPIServer` is not optional, `AuthInfo` is arguable not optional and `CloudProvider` should be optional.

WDYT?

Reply:
That sounds reasonable to me, and I agree with your assessments about the optionality of the fields.

Do you think you could add a comment to [the API doc](https://docs.google.com/document/d/1Oi9EO3Jwtp69obakl-9YpLkP764GZzsz95XJlX1a960/edit#) in case people want to discuss it further there?